Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.Control.cs

```diff
@@ -1432,39 +1449,35 @@ public virtual LayoutEngine LayoutEngine
+    /// </returns>
+    /// <param name="method">A delegate to a method that takes no parameters. </param><exception cref="T:System.InvalidOperationException">No appropriate window handle can be found.</exception><filterpriority>2</filterpriority><PermissionSet><IPermission class="System.Security.Permissions.EnvironmentPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.SecurityPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Flags="UnmanagedCode, ControlEvidence"/><IPermission class="System.Diagnostics.PerformanceCounterPermission, System, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/></PermissionSet>
+    public IAsyncResult BeginInvoke(Delegate method)
+    {
+        Contract.Requires(method != null);
+        return default(IAsyncResult);
+    }
+
+    /// <summary>
+    /// Executes the specified delegate asynchronously with the specified arguments, on the thread that the control's underlying handle was created on.
+    /// </summary>
+    /// 
+    /// <returns>
+    /// An <see cref="T:System.IAsyncResult"/> that represents the result of the <see cref="M:System.Windows.Forms.Control.BeginInvoke(System.Delegate)"/> operation.
+    /// </returns>
+    /// <param name="method">A delegate to a method that takes parameters of the same number and type that are contained in the <paramref name="args"/> parameter. </param><param name="args">An array of objects to pass as arguments to the given method. This can be null if no arguments are needed. </param><exception cref="T:System.InvalidOperationException">No appropriate window handle can be found.</exception><filterpriority>2</filterpriority><PermissionSet><IPermission class="System.Security.Permissions.EnvironmentPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.SecurityPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Flags="UnmanagedCode, ControlEvidence"/><IPermission class="System.Diagnostics.PerformanceCounterPermission, System, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/></PermissionSet>    
+    public IAsyncResult BeginInvoke(Delegate method, params object[] args)
+    {
+        Contract.Requires(method != null);
+        return default(IAsyncResult);
+    }
```

Comment:
I think it's safe to add a postcondition for non-null result for both BeginInvoke variants.